www/src/components/__tests__/docs-markdown-page.js

```diff
@@ -0,0 +1,161 @@
+import React from "react"
+import { render } from "@testing-library/react"
+import { Helmet } from "react-helmet"
+import DocsMarkdownPage from "../docs-markdown-page"
+import { ThemeProvider } from "theme-ui"
+import { I18nProvider } from "@lingui/react"
+import { MDXProvider } from "@mdx-js/react"
+
+import theme from "../../../src/gatsby-plugin-theme-ui"
+
+jest.mock("gatsby-plugin-mdx", () => ({
+  MDXRenderer: () => <div />
+}));
+
+jest.mock("../../utils/sidebar/item-list", () => {
+  return {
+    itemListContributing: {},
+    itemListDocs: {},
+    getItemList: jest.fn(),
+  }
+})
+
+jest.mock("../../hooks/use-site-metadata", () => {
+  return () => {
+    return { siteUrl: "https://www.gatsbyjs.org" }
+  }
+})
+
+Object.defineProperty(window, "IntersectionObserver", {
+  writable: true,
+  value: jest.fn().mockImplementation(() => {
+    return {
+      observe: jest.fn(),
+      unobserve: jest.fn(),
+    }
+  }),
+})
+
+Object.defineProperty(window, "matchMedia", {
+  writable: true,
+  value: jest.fn().mockImplementation(query => ({
+    matches: false,
+    addListener: jest.fn(),
+    removeListener: jest.fn(),
+  })),
+})
+
+const tableOfContentsItems = [
+  {
+    url: "#section",
+    title: "section",
+  },
+]
+const page = {
+  excerpt: "excerpt",
+  timeToRead: 1,
+  fields: {
+    slug: "/docs/apis/",
+    anchor: "apis",
+  },
+  frontmatter: {
+    title: "title",
+    description: "description",
+  },
+  tableOfContents: {
+    items: tableOfContentsItems,
+  },
+  parent: {},
+}
+
+const location = {
+  pathname: "/docs/current",
+}
+
+const setup = (setupProps = {}) => {
+  const props = {
+    page,
+    location,
+    ...setupProps,
+  }
+
+  return render(
+    <I18nProvider>
+      <ThemeProvider theme={theme}>
+        <MDXProvider>
+          <DocsMarkdownPage {...props} />
+        </MDXProvider>
+      </ThemeProvider>
+    </I18nProvider>
+  )
+}
+
+it("should display table of content if there are items and is not disabled", () => {
+  const { getByText } = setup()
+
+  expect(getByText("Table of Contents")).toBeDefined()
+})
+
+it("should not display table of content if there are no items", () => {
+  const tableOfContentsItems = []
+  const { queryByText } = setup({ tableOfContentsItems })
+
+  expect(queryByText("Table of Contents")).toBeNull()
```

Comment:
Could we do this instead, for this and the other calls to `toBeNull`?

```suggestion
expect(queryByText("Table of Contents")).not.toBeDefined()
```

This way we don't get a false failure if one of these returnes `undefined` for some reason instead of `null`.

Reply:
I tried but it doesn't work 😞Checking the doc, [queryBy](https://testing-library.com/docs/dom-testing-library/api-queries#queryby) always return `null` if no element match. I could change it to `toBeFalsy` 🤔 but I think that with `toBeNull` in this case is just fine. Let me what you think!

Reply:
Oh! in that case that should be fine then. :)